u8timeseries/utils/timeseries_generation.py

```diff
@@ -4,96 +4,111 @@
+    :param start_ts: The time index of the first entry in the returned TimeSeries.
     :return: A constant TimeSeries with value 'value'.
     """
 
-    times = pd.date_range(periods=length, freq=offset, start=start_date)
+    times = pd.date_range(periods=length, freq=freq, start=start_ts)
     values = np.full(length, value)
 
     return TimeSeries.from_times_and_values(times, values)
 
 
-def linear_timeseries(start_value: float = 0, value_delta: float = 1, length: int = 10, offset: str = 'D',
-                      start_date: pd.Timestamp = pd.Timestamp('2000-01-01')) -> 'TimeSeries':
+def linear_timeseries(start_value: float = 0, end_value: float = 1, length: int = 10, freq: str = 'D',
+                      start_ts: pd.Timestamp = pd.Timestamp('2000-01-01')) -> 'TimeSeries':
     """
-    Creates a TimeSeries with a starting value of 'start_value' that increases by 'value_delta' at each step.
-    The last entry of the time series will be equal to 'start_value' + ('length' - 1) * 'value_delta'.
+    Creates a TimeSeries with a starting value of 'start_value' that increases linearly such that
+    it takes on the value 'end_value' at the last entry of the TimeSeries. This means that
+    the difference between two adjacent entries will be equal to 
+    ('end_value' - 'start_value') / ('length' - 1).
 
     :param start_value: The value of the first entry in the TimeSeries.
-    :param value_delta: The difference in value between to adjacent entries in the TimeSeries.
+    :param end_value: The value of the last entry in the TimeSeries.
     :param length: The length of the returned TimeSeries.
-    :param offset: The time differene between two adjacent entries in the returned TimeSeries. A DateOffset alias is expected.
-    :param start_date: The time index of the first entry in the returned TimeSeries.
-    :return: A linear TimeSeries with gradient 'value_delta'.
+    :param freq: The time difference between two adjacent entries in the returned TimeSeries. A DateOffset alias is expected.
+    :param start_ts: The time index of the first entry in the returned TimeSeries.
+    :return: A linear TimeSeries created as indicated above.
     """
 
-    times = pd.date_range(periods=length, freq=offset, start=start_date)
-    values = np.linspace(start_value, start_value + (length - 1) * value_delta, length)
+    times = pd.date_range(periods=length, freq=freq, start=start_ts)
+    values = np.linspace(start_value, end_value, length)
 
     return TimeSeries.from_times_and_values(times, values)
 
 
-def periodic_timeseries(frequency: float = 0.1, amplitude: float = 1, phase: float = 0, y_offset: float = 0,
-                        length: int = 10, offset: str = 'D',
-                        start_date: pd.Timestamp = pd.Timestamp('2000-01-01')) -> 'TimeSeries':
+def periodic_timeseries(value_frequency: float = 0.1, value_amplitude: float = 1, value_phase: float = 0, 
```

Comment:
Here I think we could even name it `sine_timeseries`as not all periodic time series are sinusoids.

Reply:
Done